### ISSUE TITLE
Disk encryption

### ISSUE THREAD

**Issue by frelon** [rancher/elemental-toolkit#1782]
**Is your feature request related to a problem? Please describe.**
It should be possible to deploy a fully encrypted system using elemental.

**Describe the solution you'd like**
I have a POC from hackweek for the install-command here: https://github.com/frelon/elemental-cli/tree/cryptsetup which adds configuration options to run cryptsetup on `elemental install` to encrypt partitions.

After that there needs to be changes to the grub-config and immutable-rootfs dracut module in order to actually unlock the partitions. This will probably depend on #1781 being merged.

**Comment by kkaempf** [comment]
Disk encryption is useful to prevent against unauthorized access to data in case a machine gets physically compromised.

Not sure how this is supposed to work in an edge/cluster scenario without anyone to enter the password.

'Supporting' this in Elemental might send the wrong message 🤔

**Comment by frelon** [comment]
> Disk encryption is useful to prevent against unauthorized access to data in case a machine gets physically compromised.

Yep, edge seems the perfect candidate for encryption, since those machines are often more physically available.

> Not sure how this is supposed to work in an edge/cluster scenario without anyone to enter the password.

It's up to the user, but there are lots of ways of unlocking volumes without a password (TPM comes to mind).

Not having disk encryption at the edge feels like an oversight, but I'm happy to have a discussion about this! :+1:

**Comment by kkaempf** [comment]
@agracey 👆🏻

**Comment by Sporesirius** [comment]
NBDE (Network-Bound Disk Encryption) comes to mind. I think it is mainly used in [openshift](https://docs.openshift.com/container-platform/4.13/security/network_bound_disk_encryption/nbde-about-disk-encryption-technology.html), but the server and client are open source, [clevis](https://github.com/latchset/clevis) and [tang](https://github.com/latchset/tang).

**Comment by kkaempf** [comment]
https://lists.opensuse.org/archives/list/factory@lists.opensuse.org/thread/WGWLTHVL4EFKL2DUJIG4RMM46RS3SWKD/ !

**Comment by jmpolom** [comment]
elemental should be able to configure LUKS volumes for any partitions configured (state, persistent, oem, user defined, etc.) but currently appears not to be able to do so. These LUKS volumes should be configurable to unlock via binding to a host security device (like a TPM 2.0 or FIDO2 token) as well as a manual text input method (recovery key) using systemd-cryptenroll.

Any status on when such a feature might be supported? The proof of concept looks like a good start.

**Comment by kkaempf** [comment]
It's actually next on the list to investigate after we're done with SELinux 😉

**Comment by jmpolom** [comment]
I see this was moved to blocked a bit ago. What is the specific blocker to implementing support for encrypted volumes in elemental? Looking to see if I can contribute a resolution to this issue or if it is indeed blocked.
